### ISSUE TITLE
How do you select (with keyboard) in read-only zone?

### ISSUE THREAD

**Issue by ardumont** [mooz/keysnail#143]
Hi,

What's the right way to trigger the selection of text to copy/kill region of text in read-only zone from keysnail?

I'd like to be able to trigger the region-mode with C-space and then use C-n/C-p to select next/previous lines (the same as conkeror).

The current natural keysnail (on my machines) forces me to use arrow instead of C-n/C-p to select next/previous lines.

Did I miss something?

To make this work as I wanted, I added this to my personal configuration (which seems kind of hacky :D):

``` javascript
key.quitKey              = "C-2";

ext.add("activate-caret-mode",
        function (ev, arg) {
            setCaretMode(true);
        }, M({ja: "activate-caret-mode", en: "activate-caret-mode"}));

ext.add("deactivate-caret-mode",
        function (ev, arg) {
            setCaretMode(false);
        }, M({ja: "deactivate-caret-mode", en: "deactivate-caret-mode"}));

ext.add("status-caret-mode",
        function (ev, arg) {
            display.echoStatusBar("Caret mode " + (isCaretMode() ? "a" : "dea") + "ctivated!", 2000);
        }, M({ja: "status-caret-mode", en: "status-caret-mode"}));

function isCaretMode() {
    return util.getBoolPref("accessibility.browsewithcaret");
}

function setCaretMode(b) {
    util.setBoolPref("accessibility.browsewithcaret", b);
    display.echoStatusBar("Caret mode " + (isCaretMode() ? "a" : "dea") + "ctivated!", 2000);
}

key.setGlobalKey("C-g", function (ev) {
    setCaretMode(false);
    tmpTarget=ev.target || ev.originalTarget;
    tmpTarget.dispatchEvent(key.stringToKeyEvent(key.quitKey));
}, "Replay macro recording");

key.setEditKey([['C-SPC'], ['C-@']], function (ev) {
    // closing the find bar
    command.closeFindBar();
    // activate the caret mode if not already enabled
    setCaretMode(true);
    // at last activate the mark
    command.setMark(ev);
}, 'Set the mark', true);

key.setGlobalKey('M-w', function (ev) {
    command.copyRegion(ev);
    setCaretMode(false);
}, 'Copy selected text', true);

key.setGlobalKey('C-s', function (ev) {
    setCaretMode(true);
    command.iSearchForwardKs(ev);
}, 'Emacs like incremental search forward', true);

key.setGlobalKey('C-r', function (ev) {
    setCaretMode(true);
    command.iSearchBackwardKs(ev);
}, 'Emacs like incremental search backward', true);

```

(I tried at first to use the KeyBoardQuit hook to set the caret mode to false but did not succeed)

In short, this triggers the firefox's "accessibility.browsewithcaret", permits the selection like I want, and deactivate this mode afterward. The quit function triggers also the caret mode deactivation.

Do you know a better alternative?

Again, thanks for keysnail.

Cheers,
